include/unifex/task.hpp

```diff
@@ -216,13 +229,16 @@ struct _task<T>::type {
         coro::coroutine_handle<Promise> h) noexcept {
       assert(coro_);
       coro_.promise().continuation_ = h;
+      coro_.promise().doneCallback_ = [](void* p) noexcept {
+        return coro::coroutine_handle<Promise>::from_address(p).promise().unhandled_done();
+      };
```

Comment:
@ericniebler, is this another case where the lambda has an unnecessary implicit type dependency?  Looks to me like we really only need `Promise` here, but implicitly also depend on `T`.  Perhaps there should be a namespace-scope free function template that does this work; something like

```
namespace unifex::_task {
template <typename Promise>
coro::coroutine_handle<> task_done_callback(void* p) noexcept {
  return coro::coroutine_handle<Promise>::from_address(p).promise().unhandled_done();
}
```

Then this line could be
```suggestion
      coro_.promise().doneCallback_ = task_done_callback<Promise>;
```

Reply:
I've added in the namespace-scope function.